File: LeetCode/LeetCode 75/Maximum Twin Sum of a Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int>v;
        long long mx=0,sum=0;
        while(head)
        {
            v.push_back(head->val);
            head=head->next;
        }
        int n=v.size()-1;
        int i, l=n/2;
        for(i=0;i<=l;i++)
        {
            mx=std::max(mx,(long long)(v[i]+v[n-i]));
        }
        return (int)mx;
    }
};

File: LeetCode/LeetCode 75/Maximum Twin Sum of a Linked List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int PairSum(ListNode head) {
        List<int>l=new List<int>();
        long mx=0;
        while(head!=null)
        {
            l.Add(head.val);
            head=head.next;
        } 
        int n=l.Count-1;
        int i;
        for(i=0;i<=n/2;i++)
        {
            mx=Math.Max(mx,(long)(l[i]+l[n-i]));
        }
        return (int)mx;
    }
}
